lib/forcegraph.ts

```diff
@@ -42,64 +54,69 @@ export const ForceGraph = function (linkScale, sidebar) {
     draw.setMaxArea(canvas.width, canvas.height);
   }
 
-  function transformPosition(p) {
+  function transformPosition(p: { k: number; x: number; y: number }) {
+    // @ts-ignore
     transform.x = p.x;
+    // @ts-ignore
     transform.y = p.y;
+    // @ts-ignore
     transform.k = p.k;
   }
 
-  function moveTo(callback, forceMove) {
+  function moveTo(callback: () => number[], forceMove?: boolean) {
     clearTimeout(movetoTimer);
-    if (!forceMove && force.alpha() > 0.3) {
+    if (!forceMove && force && force.alpha() > 0.3) {
       movetoTimer = setTimeout(function timerOfMoveTo() {
         moveTo(callback);
       }, 300);
       return;
     }
-    var result = callback();
-    var x = result[0];
-    var y = result[1];
-    var k = result[2];
-    var end = { k: k };
+    let result = callback();
+    let x = result[0];
+    let y = result[1];
+    let k = result[2];
+    let end = {
+      k: k,
+      x: (canvas.width + sidebar.getWidth()) / 2 - x * k,
+      y: canvas.height / 2 - y * k,
+    };
 
-    end.x = (canvas.width + sidebar.getWidth()) / 2 - x * k;
-    end.y = canvas.height / 2 - y * k;
+    let start = { x: transform.x, y: transform.y, k: transform.k };
 
-    var start = { x: transform.x, y: transform.y, k: transform.k };
+    let interpolate = d3Interpolate.interpolateObject(start, end);
 
-    var interpolate = d3Interpolate.interpolateObject(start, end);
-
-    var timer = d3Timer.timer(function (t) {
+    let timer = d3Timer.timer(function (t) {
       if (t >= ZOOM_ANIMATE_DURATION) {
         timer.stop();
         return;
       }
 
-      var v = interpolate(d3Ease.easeQuadInOut(t / ZOOM_ANIMATE_DURATION));
+      let v = interpolate(d3Ease.easeQuadInOut(t / ZOOM_ANIMATE_DURATION));
       transformPosition(v);
       window.requestAnimationFrame(redraw);
     });
   }
 
-  function onClick(event) {
+  function onClick(event: Event & ClientPointEvent) {
     if (event.defaultPrevented) {
       return;
     }
 
-    var e = transform.invert([event.clientX, event.clientY]);
-    var node = force.find(e[0], e[1], NODE_RADIUS_SELECT);
+    let click = transform.invert([event.clientX, event.clientY]);
+    let point = { x: click[0], y: click[1] };
+    let node = force.find(point.x, point.y, NODE_RADIUS_SELECT);
+    let router = window.router;
 
     if (node !== undefined) {
+      // @ts-ignore
```

Comment:
Is `node` actually something different here?

Reply:
Its should be some kind of MapNode that has an `o` attribute with the actual node